README.md

```diff
@@ -352,476 +339,1080 @@ You can use the aws module to audit AWS resources, launch autoscaling groups in
-*Optional* A list of termination policies to use when scaling in instances. For valid termination policies, see [Controlling Which Instances Auto Scaling Terminates During Scale In](http://docs.aws.amazon.com/autoscaling/latest/userguide/as-instance-termination.html).
+Valid values are:
 
-#####`tags`
-*Optional* The tags to assign to the autoscaling group. Accepts a 'key => value' hash of tags. The tags are not propagated to launched instances.
+Valid IPv4 address.
 
-#### Type: ec2_elastic_ip
+#####`associate_public_ip_address`
 
-#####`ensure`
-Specifies that basic state of the resource. Valid values are 'attached', 'detached'.
+Optional. 
 
-#####`name`
-*Required* The IP address of the Elastic IP. Accepts a valid IPv4 address of an already existing elastic IP.
+Whether to assign a public interface in a VPC. 
 
-#####`region`
-*Required* The region in which the Elastic IP is found. For valid values, see [AWS Regions](http://docs.aws.amazon.com/general/latest/gr/rande.html#ec2_region).
+This parameter is set at creation only; it is not affected by updates. 
 
-#####`instance`
-*Required* The name of the instance associated with the Elastic IP. This is the value of the AWS Name tag.
+Valid values are: `true`, `false`. 
 
+Default value: `false`.
 
-#### Type: ec2_launchconfiguration
+#####`subnet`
 
-#####`name`
-*Required* The name of the launch configuration. This is the value of the AWS Name tag.
+Optional. 
 
-#####`security_groups`
-*Required* The security groups to associate with the instances. This parameter is set at creation only; it is not affected by updates. Accepts an array of security group names, i.e., the Name tags on the security groups.
+The VPC subnet to attach the instance to. 
 
-#####`user_data`
-*Optional* User data script to execute on new instances. This parameter is set at creation only; it is not affected by updates.
+This parameter is set at creation only; it is not affected by updates. 
+
+Accepts the name of the subnet; this is the value of the Name tag for the subnet. If you're describing the subnet in Puppet, then this value is the name of the resource.
+
+#####`ebs_optimized`
+
+Optional. 
+
+Whether or not to use optimized storage for the instance.  
+
+This parameter is set at creation only; it is not affected by updates. 
+
+Valid values are: `true`, `false`. 
+
+Default value: `false`.
+
+#####`instance_initiated_shutdown_behavior`
+
+Optional. 
+
+Whether the instance stops or terminates when you initiate shutdown from the instance. 
+
+This parameter is set at creation only; it is not affected by updates. 
+
+Valid values are: 'stop', 'terminate'. 
+
+Default value: 'stop'.
+
+#####`block_devices`
+
+Optional. 
+
+A list of block devices to associate with the instance. 
+
+This parameter is set at creation only; it is not affected by updates. 
+
+Accepts an array of hashes with the 'device name', 'volume size', 'delete on termination flag', and 'volume type' specified:
+
+``` Puppet
```

Comment:
haha, it highlighted all of that because I didn't close my language highlighting.